Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -98,23 +98,76 @@ int VertexLoaderX64::ReadVertex(OpArg data, u64 attribute, int format, int count
 
 	int elem_size = 1 << (format / 2);
 	int load_bytes = elem_size * count_in;
-	if (load_bytes > 8)
-		MOVDQU(coords, data);
-	else if (load_bytes > 4)
-		MOVQ_xmm(coords, data);
-	else
-		MOVD_xmm(coords, data);
-
-	PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
 
-	if (format != FORMAT_FLOAT)
+	if (cpu_info.bSSSE3)
 	{
-		// Sign extend
+		if (load_bytes > 8)
+			MOVDQU(coords, data);
+		else if (load_bytes > 4)
+			MOVQ_xmm(coords, data);
+		else
+			MOVD_xmm(coords, data);
+
+		PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
+
+		// Sign-extend.
 		if (format == FORMAT_BYTE)
 			PSRAD(coords, 24);
 		if (format == FORMAT_SHORT)
 			PSRAD(coords, 16);
+	}
+	else
+	{
+		switch (format)
+		{
+		case FORMAT_UBYTE:
+		case FORMAT_BYTE:
+			MOVD_xmm(coords, data);
+			PXOR(XMM1, R(XMM1));
+			PUNPCKLBW(coords, R(XMM1));
+			if (count_in > 1)
+				PUNPCKLWD(coords, R(XMM1));
+			if (format == FORMAT_BYTE)
+			{
+				PSLLD(coords, 24);
+				PSRAD(coords, 24);
+			}
+			break;
+		case FORMAT_USHORT:
+		case FORMAT_SHORT:
+			LoadAndSwap(64, scratch3, data);
+			MOVQ_xmm(coords, R(scratch3));
+			PSHUFLW(coords, R(coords), 0xE7); // ABCD EFGH -> ABCD EFGE
+			if (count_in == 3)
+			{
+				SHR(64, R(scratch3), Imm8(16));
+				PINSRW(coords, R(scratch3), 4);
+			}
+			PSLLD(coords, 16);
+			if (format == FORMAT_SHORT)
+				PSRAD(coords, 16);
+			else
+				PSRLD(coords, 16);
+			break;
+		case FORMAT_FLOAT:
+			LoadAndSwap(64, scratch3, data);
+			MOVQ_xmm(coords, R(scratch3));
+			if (count_in == 3)
+			{
+				data.offset += 8;
+				LoadAndSwap(32, scratch3, data);
+				MOVD_xmm(XMM1, R(scratch3));
+			}
+			if (count_in > 1)
+				PSHUFD(coords, R(coords), 0xE1); // swap lower 2 dwords
```

Comment:
I'll just add a GPR-only shortcut for `format == FORMAT_FLOAT && count_in == 1`.